### PR TITLE
Add warning to clarify usage of `spatial_neighbors`

### PR DESCRIPTION
**Pull request by MxMstrmn**
**IMPORTANT: Please search among the [Pull requests](../pulls) before creating one.**

## How has this been tested?

Add a warning to clarify the usage of :func:`squidpy.gr.spatial_neighbors`.

## Closes

#498

### CODE REVIEW

**Comment by giovp** [comment]
🙏 🙏 🙏

**Review by giovp — APPROVED** [review]

File: squidpy/gr/_build.py [hunk 1]
```diff
@@ -95,6 +95,11 @@ def spatial_neighbors(
 
     transform = Transform.NONE if transform is None else Transform(transform)
     if coord_type is None:
+        if radius is not None:
+            warnings.warn(
```
Comment (michalk8) [inline squidpy/gr/_build.py:99]: Please use `logg.warning` so that it's controlled via `scanpy.verbosity`

File: squidpy/gr/_build.py [hunk 2]
```diff
@@ -95,6 +95,11 @@ def spatial_neighbors(
...
+                f"\nGraph creation with `radius` is only available when `coord_type = {CoordType.GENERIC!r}` specified. "
```
Comment (michalk8) [inline squidpy/gr/_build.py:100]: Please remove unnecessary `\n`; this should also fix the linter 🤞

**Review by michalk8 — CHANGES_REQUESTED** [review]

File: squidpy/gr/_build.py (hunk 1 above)
Comment (MxMstrmn) [reply squidpy/gr/_build.py]: Done

**Review by michalk8 — APPROVED** [review]

**Pull request merged**